### DISCUSSION TITLE
Using patches to reference network interface id for VM creation

### DISCUSSION THREAD

**Discussion by vishnujanardhanan** [crossplane/crossplane#3101] [Q&A]
Hi,
I'm trying to create a Azure VM using terrajet and I'm facing issue while patching spec.forProvider.networkInterfaceIds.

I'm setting the network interface id into the XRD via the below patch

  - type: ToCompositeFieldPath
    fromFieldPath: status.atProvider.id
    toFieldPath: status.networkInterfaceId
and trying to retrieve it in the VM composition using the below patch

  patches:
  - type: FromCompositeFieldPath
    fromFieldPath: "status.networkInterfaceId"
    toFieldPath: "spec.forProvider.networkInterfaceIds[0]"

But i always end up with an error like this -> 20m Warning ComposeResources compositet0service/azt0services-8z4xb cannot render composed resource from resource template at index 4: cannot use dry-run create to name composed resource: LinuxVirtualMachine.compute.azure.jet.crossplane.io "azt0services-8z4xb-8dbgm" is invalid: spec.forProvider.networkInterfaceIds: Required value

I have the definition and composition in my git repo -> https://github.com/vishnujanardhanan/crossplane-composition-poc/blob/main/az-composition.yaml

Please help me

**Comment by garreeoke** [comment]
@vishnujanardhanan 

Was the status.atProvider.id field set in the status of the managed resource? Was the status.networkInterfaceId set on the XR?  The warning message you see above should reconcile once there is a value in these status fields.
